libvast/src/detail/load_plugin.cpp

```diff
@@ -124,4 +124,113 @@ load_plugin(const std::filesystem::path& path_or_name,
                                      fmt::join(load_errors, "\n - ")));
 }
 
+caf::expected<std::vector<std::filesystem::path>>
+load_plugins(std::vector<std::string> bundled_plugins,
+             caf::actor_system_config& cfg) {
+  auto loaded_plugin_paths = std::vector<std::filesystem::path>{};
+  auto paths_or_names
+    = caf::get_or(cfg, "vast.plugins", std::vector<std::string>{});
+  if (paths_or_names.empty())
+    return loaded_plugin_paths;
+  auto& plugins = plugins::get();
+  auto load_specified_plugins = [&]() -> caf::error {
+    for (const auto& plugin_path_or_name : paths_or_names) {
+      if (auto loaded_plugin = load_plugin(plugin_path_or_name, cfg)) {
+        auto&& [path, plugin] = std::move(*loaded_plugin);
+        loaded_plugin_paths.push_back(std::move(path));
+        plugins.push_back(std::move(plugin));
```

Comment:
I know out-references are a code smell, but *implicit* out-references are surely even smellier.

Reply:
I renamed `loaded_plugin_paths` to `result` to make that clearer, and replaced `plugins` with the mutable singleton.